src/components/appointments/appointment-drawer/activity-log-modal/helpers.ts

```diff
@@ -0,0 +1,108 @@
+import { format, getHours, getMinutes, isToday, isWithinInterval, parseISO } from 'date-fns';
+import { utcToZonedTime } from 'date-fns-tz';
+import { APPOINTMENT_STATUS, APPOINTMENT_TYPE, USER_ROLE } from 'src/constants';
+import { DetailedAppointment } from 'src/redux/api/appointmentApi';
+import { UserRole } from 'src/redux/slices/userSlice';
+import { UTC_BIAS, WEEKDAY_FORMAT } from './constants';
+
+const getISODateWithoutUTC = (date: string): string => date.replace(UTC_BIAS, '');
+
+const isRecurringWorkingDay = (now: Date, weekday: string[]): boolean =>
+  weekday.includes(format(now, WEEKDAY_FORMAT));
+
+const isWithinAppointmentInterval = (now: Date, startDate: string, endDate: string): boolean => {
+  const start = parseISO(startDate);
+  const end = parseISO(endDate);
+
+  return isWithinInterval(now, { start, end });
+};
+
+export const isActivityLogShown = (appointment: DetailedAppointment): boolean => {
+  const { startDate, endDate, status, timezone, type, weekday } = appointment;
+
+  if (type === APPOINTMENT_TYPE.OneTime && status === APPOINTMENT_STATUS.Completed) {
+    return true;
+  }
+
+  if (
+    type === APPOINTMENT_TYPE.Recurring &&
+    (status === APPOINTMENT_STATUS.Completed || status === APPOINTMENT_STATUS.Active)
+  ) {
+    const now = utcToZonedTime(new Date(), timezone);
+    const { isWorkingDay, hasPassedStartTime, hasPassedEndTime } = calculateDateInfo(
+      now,
+      startDate,
+      endDate,
+      weekday as string[]
+    );
+
+    return (
+      isWorkingDay &&
+      isWithinAppointmentInterval(now, startDate, endDate) &&
+      (hasPassedEndTime || !hasPassedStartTime)
+    );
+  }
+
+  return false;
+};
+
+const calculateDateInfo = (
```

Comment:
in each helper add input and output example in comments. are you sure that all helper functions are used only for this component and could not be reused somewhere else?

Reply:
fixed, pulled one of the helper functions to the globals because it can be reused later, other functions contain logic specific only for this part of drawer